python/utils/func_utils.py

```diff
@@ -48,4 +52,157 @@ def __call__(cls, *args, **kwargs):
         if cls not in cls._instances:
             cls._instances[cls] = super(
                 Singleton, cls).__call__(*args, **kwargs)
-        return cls._instances[cls]
\ No newline at end of file
+        return cls._instances[cls]
+
+class PropertyModel(BaseModel):
+    """参数属性模型，用于描述函数参数的类型和元数据。
+
+    Attributes:
+        type (Any): 参数的类型。
+    """
+    type: Any
+
+
+class ParametersModel(BaseModel):
+    """函数参数模型，用于定义函数的参数结构。
+
+    Attributes:
+        type (Literal["object"]): 表示参数集合的类型，固定为 "object"。
+        properties (Dict[str, PropertyModel]): 参数的具体属性映射，其中键是参数名，值是对应的属性模型。
+        required (List[str]): 必须提供的参数列表。
+    """
+    type: Literal["object"]
```

Comment:
Use explicit name, description, type, required.  Leave properties for extension.